--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -58,10 +58,10 @@
 endif()
 
 ## Plugins
-add_subdirectory(plugins/sceneparsers)
-add_subdirectory(plugins/geometryloaders)
-add_subdirectory(plugins/renderers)
+#add_subdirectory(plugins/sceneparsers)
+#add_subdirectory(plugins/geometryloaders)
+#add_subdirectory(plugins/renderers)
 
 if(TARGET Qt::Quick)
-    add_subdirectory(plugins/renderplugins)
+#    add_subdirectory(plugins/renderplugins)
 endif()

