--- a/pdf_viewer_build_config.pro
+++ b/pdf_viewer_build_config.pro
@@ -6,7 +6,8 @@
               zlib
           
 
-QT += core opengl gui widgets network 3dinput quickwidgets svg texttospeech
+QT += core opengl gui widgets network quickwidgets svg texttospeech waylandclient-private
+QTPLUGIN += qwayland-generic qwayland-egl
 
 greaterThan(QT_MAJOR_VERSION, 5){
 	QT += openglwidgets
@@ -17,7 +18,7 @@
 }
 
 CONFIG += c++17
-DEFINES += QT_3DINPUT_LIB QT_OPENGL_LIB QT_OPENGLEXTENSIONS_LIB QT_WIDGETS_LIB
+DEFINES += QT_OPENGL_LIB QT_OPENGLEXTENSIONS_LIB QT_WIDGETS_LIB
 
 CONFIG(non_portable){
     DEFINES += NON_PORTABLE
@@ -88,7 +89,7 @@
     QMAKE_CXXFLAGS += -std=c++17
 
     CONFIG(linux_app_image){
-        LIBS += -ldl -Lmupdf/build/release -lmupdf -lmupdf-third -lmupdf-threads -lharfbuzz -lz
+        LIBS += -ldl -Lmupdf/build/release -lmupdf -lmupdf-third -lmupdf-threads -lharfbuzz -lz -lfreetype -lopenjp2 -ljpeg
     } else {
         DEFINES += NON_PORTABLE
         DEFINES += LINUX_STANDARD_PATHS
@@ -96,7 +97,7 @@
     }
 
     isEmpty(PREFIX){
-        PREFIX = /usr
+        PREFIX = /app
     }
     target.path = $$PREFIX/bin
     shortcutfiles.files = resources/sioyek.desktop
--- a/build_linux.sh
+++ b/build_linux.sh
@@ -3,7 +3,7 @@
 
 # Compile mupdf
 cd mupdf
-make USE_SYSTEM_HARFBUZZ=yes -j$(nproc)
+make USE_SYSTEM_HARFBUZZ=yes USE_SYSTEM_FREETYPE=yes USE_SYSTEM_OPENJPEG=yes USE_SYSTEM_ZLIB=yes USE_SYSTEM_LIBJPEG=yes -j$(nproc)
 cd ..
 
 # Compile sioyek
